backend/courses/filters.py

```diff
@@ -313,16 +313,17 @@ def filter_choices(queryset, choices):
 
     return filter_choices
 
+
 def degree_rules_filter(queryset, rule_ids):
     """
     :param queryset: initial Course object queryset
     :param rule_ids: Comma separated string of of Rule ids to filter by. If the rule does not
-        have a q object, it does not filter the queryset. 
+        have a q object, it does not filter the queryset.
     """
     if not rule_ids:
         return queryset
     query = Q()
-    for rule_id in rule_ids.split(","): 
+    for rule_id in rule_ids.split(","):
         try:
             rule = Rule.objects.get(id=int(rule_id))
```

Comment:
pref: `Rule.objects.filter(id=int(rule_id))` instead of try catch. 
also can use walrus operator.

`if not (rule := Rule.objects.filter(id=int(rule_id)).first()): continue`
`if not (q := rule.get_q_object()): continue`